### PR TITLE
[SPARK-30335][SQL][DOCS] Add a note first, last, collect_list and collect_set can be non-deterministic in SQL function docs as well

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?
This PR adds a note first and last can be non-deterministic in SQL function docs as well.
This is already documented in `functions.scala`.

### Why are the changes needed?
Some people look reading SQL docs only.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Jenkins will test.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/First.scala [hunk 1]
```diff
@@ -43,6 +43,10 @@ import org.apache.spark.sql.types._
       > SELECT _FUNC_(col, true) FROM VALUES (NULL), (5), (20) AS tab(col);
        5
   """,
+  note = """
+    The function is non-deterministic because its results depends on order of rows which
```
Comment (yaooqinn) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/First.scala]: depend on the order of rows

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -549,7 +549,7 @@ object functions {
    * The function by default returns the last values it sees. It will return the last non-null
    * value it sees when ignoreNulls is set to true. If all values are null, then null is returned.
    *
-   * @note The function is non-deterministic because its results depends on order of rows which
+   * @note The function is non-deterministic because its results depends on the order of rows which
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: Total nit, but you use "on the order or rows" here but "on order of the rows" above. The most correct would be 'the order of the rows' but honestly doesn't matter much. Feel free to standardize on anything

Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: Oops

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thanks @dongjoon-hyun, @yaooqinn, @srowen.

Merged to master.

**Pull request closed without merging**